### PR TITLE
NOT FOR MERGE Ophcrack 

### PR DESCRIPTION
**Pull request by kirillrdy**
## Description of changes

refactoring of https://github.com/NixOS/nixpkgs/pull/290496

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/ophcrack/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          "--enable-gui"
+          "--with-qt5charts"
+        ]
+      else
+        [ "--disable-gui" ]
+    );
+
+  postinstall = lib.optional enableGui ''
+    wrapProgram $out/bin/ophcrack \
+      --prefix PATH : "${libsForQt5.qtbase}/bin"
```
Comment (kirillrdy) [inline pkgs/by-name/op/ophcrack/package.nix:50]: @Tochiaha can you explain why this is necessary ?

File: pkgs/by-name/op/ophcrack/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  expat,
+  enableGui ? true,
+}:
+stdenv.mkDerivation {
+
+  pname = "ophcrack";
+  version = "3.8.0";
+
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/ophcrack/files/ophcrack/3.8.0/ophcrack-3.8.0.tar.bz2";
```
Comment (kirillrdy) [inline pkgs/by-name/op/ophcrack/package.nix:18]: we should use version here

File: pkgs/by-name/op/ophcrack/package.nix (hunk 1 above)
Comment (Tochiaha) [reply pkgs/by-name/op/ophcrack/package.nix:50]: with this nix expression their wont be any need for the postinstall. i was experiencing this [error](https://discourse.nixos.org/t/appimage-run-fails-loading-qt-platform-plugin-xcb/29293?u=tochi4h4) before this solution.

Comment (kirillrdy) [reply pkgs/by-name/op/ophcrack/package.nix:50]: how do you reproduce it, just run it ? I don't get any errors without this `postinstall`

I am on gnome wayland

Comment (Tochiaha) [reply pkgs/by-name/op/ophcrack/package.nix:50]: i don't get any errors without the postInstall now.

Comment (kirillrdy) [reply pkgs/by-name/op/ophcrack/package.nix:50]: if you don't have any questions, feel free to cherry pick this to your PR

**Pull request closed without merging**